aggregator/src/aggregator/collection_job_driver.rs

```diff
@@ -376,9 +376,7 @@ impl CollectionJobDriver {
                 Ok((Method::PUT, Some(request_body)))
             }
             CollectionJobState::Poll => Ok((Method::GET, None)),
-            // TODO: This needs to send DELETE, not GET, but our Helper isn't able to do this
-            // yet. See Issue #3962
-            CollectionJobState::Deleted => Ok((Method::GET, None)),
+            CollectionJobState::Deleted => Ok((Method::DELETE, None)),
```

Comment:
What happens if the DELETE the leader sends to the helper fails? Presumably we'll do the usual retry dance, but if we give up on that, what state is the collection job left in on the leader side?

Reply:
@tgeoghegan Take a look at the update in 438bd575